### PR TITLE
feat: add features for fine-tuning

### PR DESCRIPTION
**Pull request by maw501**
Set-up and configs for fine-tuning experiments done at 1B. In particular:

- [x] Add README for conducting experiments.
- [x] Set-up Weights and Biases to log to `chemnlp` team page [here](https://stability.wandb.io/chemnlp/LLCheM). 
- [x] Add config for 1B model.
- [x] Add ability to load from checkpoint. 
- [x] Add saving of model at end of training.
- [x] Minor config changes to support partial epoch training. 
- [x] Add file to calculate NLL for different models.

### CODE REVIEW

**Comment by maw501** [comment]
I'm not sure what's going on with `pre-commit` in the CI - I had to bypass it in order to get the remote pipeline to pass. 

Locally it wants to modify the imports for `run_tune.py` using `isort`  but when these changes are pushed it then fails the remote CI. Both local and remote CI are using `isort=5.12.0`. 🤷

Example: note it's only `import wandb` which moves.

Fails (`isort` modifies `run_tune.py` to this):

```python
import argparse
import os

import datasets
import transformers
from peft import PromptTuningConfig, PromptTuningInit, TaskType, get_peft_model
from transformers import (
    AutoTokenizer,
    DataCollatorForLanguageModeling,
    Trainer,
    TrainingArguments,
)

import wandb
from chemnlp.data_val.config import TrainPipelineConfig
from chemnlp.utils import load_config
```

Passes:

```python
import argparse
import os

import datasets
import transformers
import wandb
from peft import PromptTuningConfig, PromptTuningInit, TaskType, get_peft_model
from transformers import (
    AutoTokenizer,
    DataCollatorForLanguageModeling,
    Trainer,
    TrainingArguments,
)

from chemnlp.data_val.config import TrainPipelineConfig
from chemnlp.utils import load_config
```

File: docs/EXPERIMENT.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    torchrun --nnodes 1 --nproc-per-node 4 experiments/scripts/run_tune.py experiments/configs/hugging-face/full_160M.yml
+    ```
+- You can use `nvidia-smi` or `wandb` logging to monitor efficiency during this step.
+
+## Launching an experiment run through SLURM
+- Take the `sbatch_<suffix>` script associated with the training run and execute this through an `sbatch` command as shown in [the documentation](https://github.com/OpenBioML/chemnlp/tree/main/experiments). This will build the conda environment and install `chemnlp` before the job begins. Note that building the environment can be a little slow so if you aren't confident your code will run it's best to test it interactively first.
+- Example command:
+
+```bash
+sbatch experiments/scripts/sbatch_train_hf.sh experiments/maw501 maw501 160M_full.yml
```
Comment (bethanyconnolly) [inline docs/EXPERIMENT.md:25]: maybe this should be more general to remove your personal file names e.g.
sbatch {script} $1 $2 $3
$1: prefix to directory where the environment is saved
$2: directory to use when building the environment
$3: path to training config

File: docs/EXPERIMENT.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+```
+- Locally you can monitor your job at `/fsx/proj-chemnlp/experiments/logs` or as set in the `sbatch` script.
```
Comment (bethanyconnolly) [inline docs/EXPERIMENT.md:27]: This assumes the user has access to stability cluster.
Maybe change to "From within the stability cluster, you can monitor your job at `/fsx/proj-chemnlp/experiments/logs` or as set in the `sbatch` script."

File: docs/EXPERIMENT.md (hunk 1 above)
Comment (maw501) [reply docs/EXPERIMENT.md:25]: Yeah, I thought this - but I wanted to give an explicit example. I don't mind which we do, but favour whatever is clearest - I can update to try make clearer.

**Pull request merged**